src/app/view/web/WebChessView.ts

```diff
@@ -21,20 +23,47 @@ export class WebChessView implements ChessBoardView {
     this.viewEventBus.listenOn(eventType, reaction);
   }
 
-  showSelectedPiece(position: { x: number; y: number }): void {}
+  showSelectedPiece(position: Position): void {
+    const selectedSquare = this.translatePositionToSquare(position);
+    this.parent.querySelector(`#${selectedSquare.column.toLowerCase()}${selectedSquare.row}`)?.classList.add('square--selected');
+  }
+
+  hideSelection(): void {
+    this.parent.querySelectorAll('.square')?.forEach((square) => {
+      square.classList.remove('square--selected');
+    });
+  }
+
+  // te sqauresToHighlight idą do podświetlenia
+  showAvailableMoves(squaresToHighlight: string[]): void {
+    squaresToHighlight.forEach((square) => {
+      this.parent.querySelector(`#${square}`)?.classList.add('square--possibleMove');
+    });
+  }
+
+  hideAllAvailableMoves(): void {
+    this.parent.querySelectorAll('.square')?.forEach((square) => {
+      square.classList.remove('square--possibleMove');
+    });
+  }
 
   private renderPiecesOnBoard(piecesPositions: PiecesBoardPositions) {
-    Object.keys(piecesPositions)
-      .map((square) => {
-        const pieceName = piecesPositions[square].name.toLowerCase();
-        const pieceSide = piecesPositions[square].side.toLowerCase();
+    let piecesSquareId: string;
```

Comment:
To rozwiazanie ze zmiennymi zmieniajacymi sie co kazdy przebieg i petla jest bardzo nieeleganckie.
Da się to zrobić na operatorach map i forEach.